src/main/java/com/dev/cinema/controllers/ShoppingCartController.java

```diff
@@ -0,0 +1,45 @@
+package com.dev.cinema.controllers;
+
+import com.dev.cinema.model.dto.shoppingcart.ShoppingCartMapper;
+import com.dev.cinema.model.dto.shoppingcart.ShoppingCartResponseDto;
+import com.dev.cinema.service.MovieSessionService;
+import com.dev.cinema.service.ShoppingCartService;
+import com.dev.cinema.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/shoppingcarts")
+public class ShoppingCartController {
+    private final ShoppingCartService shoppingCartService;
+    private final UserService userService;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final MovieSessionService movieSessionService;
+
+    public ShoppingCartController(ShoppingCartService shoppingCartService,
+                                  UserService userService,
+                                  ShoppingCartMapper shoppingCartMapper,
+                                  MovieSessionService movieSessionService) {
+        this.shoppingCartService = shoppingCartService;
+        this.userService = userService;
+        this.shoppingCartMapper = shoppingCartMapper;
+        this.movieSessionService = movieSessionService;
+    }
+
+    @PostMapping("/addmoviesession")
+    public void add(@RequestParam(name = "movieSessionId", required = true) Long movieSessionId,
```

Comment:
```suggestion
    public void add(Long movieSessionId,
```
в тебе по суті name = "movieSessionId" і Long movieSessionId мають однакові імена. Тому параметр name в даному випадку зайвий